presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -555,6 +549,32 @@ public Expression visitFunctionCall(FunctionCall node, Context<C> context)
             return node;
         }
 
+        // Since OrderBy contains list of SortItems, we want to process each SortItem's key, which is an expression
+        private OrderBy rewriteOrderBy(OrderBy orderBy, Context<C> context)
+        {
+            ImmutableList.Builder<SortItem> sortItems = ImmutableList.builder();
+            for (SortItem sortItem : orderBy.getSortItems()) {
+                Expression sortKey = rewrite(sortItem.getSortKey(), context.get());
+                if (sortItem.getSortKey() != sortKey) {
+                    sortItems.add(new SortItem(sortKey, sortItem.getOrdering(), sortItem.getNullOrdering()));
+                }
+                else {
+                    sortItems.add(sortItem);
+                }
+            }
+            List<SortItem> rewrittenSortItems = sortItems.build();
+            if (sameElements(orderBy.getSortItems(), rewrittenSortItems)) {
+                return orderBy;
+            }
+
+            if (orderBy.getLocation().isPresent()) {
```

Comment:
Other rewrites don't preserver location in rewritten nodes. Let's keep it the same here since message with location about rewritten might be inconsistent